examples/eventsParser/eventsParser.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"github.com/xmidt-org/arrange"
+	"github.com/xmidt-org/interpreter"
+	"github.com/xmidt-org/interpreter/history"
+	"github.com/xmidt-org/interpreter/validation"
+	"go.uber.org/fx"
+)
+
+const (
+	applicationName = "eventsParser"
+)
+
+type Config struct {
+	Codex    CodexConfig
+	FilePath string
+	UseJSON  bool
+}
+
+type BootCycle struct {
+	ID       string
+	EventIDs []string
+	Err      error
+}
+
+func main() {
+	v := viper.New()
+	v.AddConfigPath(".")
+	v.SetConfigName(applicationName)
+	err := v.ReadInConfig()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to read in viper config: %v\n", err.Error())
+		os.Exit(1)
+	}
+
+	app := fx.New(
+		arrange.ForViper(v),
+		arrange.Provide(Config{}),
+		Provide(),
+		fx.Provide(
+			arrange.UnmarshalKey("codex", CodexConfig{}),
+		),
+		fx.Invoke(
+			readCommandLine,
+		),
+	)
+
+	if err := app.Err(); err == nil {
+		app.Run()
+	} else if errors.Is(err, pflag.ErrHelp) {
+		return
+	} else {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(2)
+	}
+}
+
+func readCommandLine(config Config, client *CodexClient) {
+	validators := createValidators()
+	comparator := testComparator()
+	if config.UseJSON {
+		var events []interpreter.Event
+		data, err := ioutil.ReadFile(config.FilePath)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "unable to read from file")
+			os.Exit(1)
+		}
+
+		if err = json.Unmarshal(data, &events); err != nil {
+			fmt.Fprintf(os.Stderr, "unable to unmarshal json: %v", err)
+			os.Exit(1)
+		}
+
+		bootCycles := parseIntoCycles(events, comparator, validators)
+		printBootCycles(bootCycles)
+		os.Exit(0)
+	} else {
+		scanner := bufio.NewScanner(os.Stdin)
+		for {
+			scanner.Scan()
+			id := scanner.Text()
+			if len(id) > 0 {
+				events := client.getEvents(id)
+				bootCycles := parseIntoCycles(events, comparator, validators)
+				printBootCycles(bootCycles)
+			}
+		}
```

Comment:
```suggestion
		}
		if err := scanner.Err(); err != nil {
			fmt.Fprintln(os.Stderr, "reading standard input:", err)
		}
```

We may not want this ^